apps/hash-frontend/src/pages/settings/integrations/google-sheets/create-or-edit-sheets-integration.tsx

```diff
@@ -0,0 +1,412 @@
+  done,
+  startExpanded,
+  title,
+}: PropsWithChildren<{
+  disabled?: boolean;
+  done: boolean;
+  startExpanded?: boolean;
+  title: string;
+}>) => {
+  const [expanded, setExpanded] = useState(!disabled && startExpanded);
+
+  return (
+    <Box>
+      <Stack
+        alignItems="center"
+        component="button"
+        direction="row"
+        justifyContent="space-between"
+        onClick={() => {
+          if (!disabled) {
+            setExpanded(!expanded);
+          }
+        }}
+        sx={({ palette }) => ({
+          cursor: "pointer",
+          width: "100%",
+          background: disabled ? palette.gray[50] : palette.blue[70],
+          border: "none",
+          py: 1,
+          px: 2,
+        })}
+      >
+        <Typography
+          sx={{
+            color: ({ palette }) => palette.common.white,
+            fontWeight: 600,
+            textAlign: "left",
+          }}
+        >
+          {title}
+        </Typography>
+        {done && (
+          <CheckIcon
+            sx={({ palette }) => ({ fill: palette.common.white, fontSize: 18 })}
+          />
+        )}
+      </Stack>
+
+      <Collapse in={expanded}>
+        <Box
+          sx={({ palette }) => ({
+            p: 2,
+            background: palette.gray[5],
+            border: `1px solid ${palette.gray[30]}`,
+          })}
+        >
+          {children}
+        </Box>
+      </Collapse>
+    </Box>
+  );
+};
+
+type IntegrationData = {
+  audience: "human" | "machine";
+  googleAccountId?: string;
+  query?: MultiFilter;
+  existingFile?: {
+    id: string;
+    name: string;
+  };
+  newFileName?: string;
+};
+
+type CreateOrEditSheetsIntegrationProps = {
+  close: () => void;
+  currentIntegration: UseSheetsIntegrationsData["integrations"][0] | null;
+  onComplete: () => void;
+};
+
+export const CreateOrEditSheetsIntegration = ({
+  close,
+  currentIntegration,
+  onComplete,
+}: CreateOrEditSheetsIntegrationProps) => {
+  const [accessToken, setAccessToken] = useState<string | null>(null);
+  const [showReauthModal, setShowReauthModal] = useState(false);
+
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const [integrationData, setIntegrationData] = useState<IntegrationData>({
+    audience:
+      (currentIntegration?.properties[
+        "https://hash.ai/@hash/types/property-type/data-audience/"
+      ] as "human" | "machine" | undefined) ?? "human",
+    existingFile: currentIntegration
+      ? {
+          id: currentIntegration.properties[
+            "https://hash.ai/@hash/types/property-type/file-id/"
+          ],
+          name: "[existing file - todo, get file name]",
+        }
+      : undefined,
+    googleAccountId:
+      currentIntegration?.account.properties[
+        "https://hash.ai/@google/types/property-type/account-id/"
+      ],
+    query: currentIntegration?.query.properties[
+      "https://blockprotocol.org/@hash/types/property-type/query/"
+    ] as MultiFilter | undefined,
+  });
```

Comment:
True and fair, but I'm going to leave it for now because we'll probably end up defining these integrations as part of a Flow rather than on this screen – will bear it in mind for when we have the proper designs